libethash-cuda/ethash_cuda_miner.cpp

```diff
@@ -63,6 +65,11 @@ ethash_cuda_miner::search_hook::~search_hook() {}
 
 ethash_cuda_miner::ethash_cuda_miner()
 {
+	int devicesCount = getNumDevices();
+	m_light = new hash64_t*[devicesCount];
```

Comment:
WTF? Use `std::vector<hash64_t*>(deviceCount)`.

Reply:
We only have a fixed no. of devices, i use this as a lookup whether a device has a light-struct already allocated. Since the number of devices during an execution doesn't change its save to use an array and index it by device_num

Reply:
std::vector will do the same (single allocation once) and you will not have to fill it with nullptrs manually and delete in the end.